src/app/codegen-data-model/tests/AttributeReportIBEncodeDecode.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "AttributeReportIBEncodeDecode.h"
+
+#include <app/MessageDef/AttributePathIB.h>
+#include <app/MessageDef/AttributeReportIB.h>
+
+using namespace chip::app;
+
+namespace chip {
+namespace Test {
+
+CHIP_ERROR DecodedAttributeData::DecodeFrom(const AttributeDataIB::Parser & parser)
+{
+    ReturnErrorOnFailure(parser.GetDataVersion(&dataVersion));
+
+    AttributePathIB::Parser pathParser;
+    ReturnErrorOnFailure(parser.GetPath(&pathParser));
+    ReturnErrorOnFailure(pathParser.GetConcreteAttributePath(attributePath, AttributePathIB::ValidateIdRanges::kNo));
+    ReturnErrorOnFailure(parser.GetData(&dataReader));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DecodeAttributeReportIBs(ByteSpan data, std::vector<DecodedAttributeData> & decoded_items)
```

Comment:
What's the lifetime of the vector? Do those hang around a lot? This is one of the first instances of std::vector in core code.

Reply:
Also, as a follow-up, suggest doing 2 pass read: first pass to count elements, then do a reserve on the vector, then actually process. This will reduce the amount of heap used transiently.

Reply:
This is specificaly for unit tests, so I did not try hard to optimize it. I would do so only if we actually encounter issues (which is unlikely because when these ever run on devices, they would be individually separated otherwise they cannot really run with others due to ember overrides).